### ISSUE TITLE
Re-sourcing .vimrc causes error when bundles are disabled with NeoBundleDisable

### ISSUE THREAD

**Issue by paulhybryant** [Shougo/neobundle.vim#417]
One example is Raimondi/delimitMate

When this bundle is not disabled with NeoBundleDisable,
so ~/.vimrc in vim runs fine.

However, if it is, so ~/.vimrc will cause a bunch of errors, which comes from plugin/delimitMate.vim

I am not sure why that file will be sourced, when a bundle is disabled.

**Comment by Shougo** [comment]
Please upload minimal .vimrc less than 50 lines.
I think it is feature.  Because, you must execute `:NeoBundleDisable` before `neobundle#end()`.
`neobundle#end()` loads scripts.  So, it will cause the error.

**Comment by Shougo** [comment]
Please read the documentation.

```
:NeoBundleDisable {name}...			*:NeoBundleDisable*
		Disables bundles specified by {name}.  If a bundle is
		disabled, its path will be removed from 'runtimepath'.
		Note: This command must be executed before neobundle loads
		the plugins(after |neobundle#end()|).
```

**Comment by paulhybryant** [comment]
Thanks for replying. I did read the documentation and NeoBundleDisable is put before neobundle#end() in my vimrc.

After updating NeoBundle today, I cannot reproduce this but there is a regression related.
Here is how to reproduce:
.vimrc.repro
########## BEGIN OF FILE ##########
set nocompatible
filetype off
set runtimepath+=$HOME/.vim/bundle/neobundle.vim/
call neobundle#begin('$HOME/.vim/bundle')

NeoBundleFetch 'Shougo/neobundle.vim'
NeoBundle 'Raimondi/delimitMate'
let s:delimitmate = neobundle#get('delimitMate')
function s:delimitmate.hooks.on_source(bundle)
  let g:delimitMate_expand_cr = 0
endfunction

for bundle in [ 'delimitMate' ]
  execute 'NeoBundleDisable' bundle
endfor

filetype plugin indent on
call neobundle#end()
NeoBundleCheck
########## END OF FILE ##########

vi -u .vimrc.repro
And the following error is reported:
[neobundle] delimitMate is already sourced.  Cannot be disabled.

**Comment by Shougo** [comment]
OK.  I will fix the problem.
But it will be big changes.

I will create new branch for it.

**Comment by paulhybryant** [comment]
Thanks for fixing it. I didn't see this error "[neobundle] BUNDLE_NAME is sourced, Cannot be disabled" before updating neobundle yesterday. Is this a regression? Disabling seems to be working fine before.

**Comment by Shougo** [comment]
Yes, it is my mistake.

**Comment by paulhybryant** [comment]
Thank you for fixing it.

On Sat, Sep 12, 2015 at 9:35 AM, Shougo <notifications@github.com> wrote:

> Closed #417 <https://github.com/Shougo/neobundle.vim/issues/417> via
> 4da6bc5
> <https://github.com/Shougo/neobundle.vim/commit/4da6bc5a128b5622ff4a7c99ee9b778e782e04ef>
> .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Shougo/neobundle.vim/issues/417#event-407655621>.
>



-- 
With Regards,

Yu Huang

Software Engineer,
Google Inc.
